### ISSUE TITLE
autosummary: template improvement

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#249]
if autosummary_generate = True
        then stub files should be gerenated 
        automatically for all subsequent Modules

 `
# !ReST

{{ fullname }}
{{ underline }}

.. automodule:: {{ fullname }}

   {&#37; block functions &#37;}
   {&#37; if functions &#37;}
   .. rubric:: Functions

   .. autosummary::
      :toctree: ./generated/

   {&#37; for item in functions &#37;}
      {{ item }}
   {&#37;\- endfor &#37;}
   {&#37; endif &#37;}
   {&#37; endblock &#37;}

   {&#37; block classes &#37;}
   {&#37; if classes &#37;}
   .. rubric:: Classes 

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/249
- Originally reported by: [timmie](https://bitbucket.org/timmie)
- Originally created at: 2009-08-29T17:57:50.428

**Comment by shimizukawa** [comment]
_From [timmie](https://bitbucket.org/timmie) on 2009-08-29 16:02:13+00:00_

attached the improved template. there should be a switch in the autosummary code that selects this attached template for cases where 

{{{
#!python

autosummary_generate = True
}}}

is set in the //conf.py//

**Comment by shimizukawa** [comment]
_From [pv](https://bitbucket.org/pv) on 2009-08-30 13:04:22+00:00_

The attached template adds a :toctree: option. We can't hardcode its value, although an empty value would be ok in principle here -- output would be placed where the other stub files go. A similar issue concerns also the class directive.

The problem here is that adding the :toctree: entries easily causes duplicates in the document tree. This is why it is not in by default -- only the end user knows if this will work. So at the moment, I don't think it is a good idea to put this in by default, as this adds too much magic into how the autosummary directive works -- the recursive stub generation is an advanced feature.

**Comment by shimizukawa** [comment]
_From [pv](https://bitbucket.org/pv) on 2009-08-30 13:05:49+00:00_

Of course, what can be done is that the documentation can be enhanced by adding an example. Perhaps also the templates could be modified so that it is easier to enable this.

**Comment by tk0miya** [comment]
Note: attached template (improved) is here.
```
{{ fullname }}
{{ underline }}

.. automodule:: {{ fullname }}

   {% block functions %}
   {% if functions %}
   .. rubric:: Functions

   .. autosummary::
      :toctree: ./generated/

   {% for item in functions %}
      {{ item }}
   {%- endfor %}
   {% endif %}
   {% endblock %}

   {% block classes %}
   {% if classes %}
   .. rubric:: Classes

   .. autosummary::
      :toctree: generated/

   {% for item in classes %}
      {{ item }}
   {%- endfor %}
   {% endif %}
   {% endblock %}

   {% block exceptions %}
   {% if exceptions %}
   .. rubric:: Exceptions

   .. autosummary::
      :toctree: generated/

   {% for item in classes %}
      {{ item }}
   {%- endfor %}
   {% endif %}
   {% endblock %}
```

**Comment by tk0miya** [comment]
I think this was also resolved by #6040. So closing now.
